### DISCUSSION TITLE
Can't figure out how to save plot objects as STL file.

### DISCUSSION THREAD

**Discussion by eduvillasr** [pyvista/pyvista#4916] [Q&A]
## Description

I am currently working on a research project and need to save multiple stl meshes as one file. I cannot figure out how to save a plot into an stl. Here is my code:

## Example Data

```py
for i in range(len(code)):
    if code[i] == 255:
        code[i] = coordinates[code.index(code[i])]
        index_name = (code.index(code[i]))
        index = coordinates[code.index(code[i])]
        name = spheres_list[index_name]
        name = testsphere.copy()
        name.translate(index)
    elif code[i] == 0:
        continue
    plotter.add_mesh(name,'r')
    points = name.points
    meshes.append(points)
```

I have a loop that creates a new STL model that overwrites itself as 'name.stl' I am wanting to combine all iterations of 'name.stl' into one stl file.

**Comment by akaszynski** [comment]
I'd combine all of them using [merge](https://docs.pyvista.org/core/filters.html?highlight=merge#pyvista.DataSetFilters.merge).  For example:
```py
import pyvista

mesh_a = pyvista.Sphere()
mesh_b = pyvista.Sphere(center=(2, 0, 0))
mesh_c = pyvista.Sphere(center=(-2, 0, 0))

# merge, we need extract_surface in 0.31.3, will be fixed
merged = mesh_a.merge([mesh_b, mesh_c]).extract_surface()
merged.save('merged.stl')
```
